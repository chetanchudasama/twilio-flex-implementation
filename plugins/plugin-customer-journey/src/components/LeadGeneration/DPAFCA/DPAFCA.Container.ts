import { connect } from "react-redux";
import { Dispatch } from "redux";
import { CustomerDetailModel, Shared } from "@common/components";

import { AppState, namespace } from "../../../states";
import { DPAFCA } from "./DPAFCA";
import { StateToProps, DispatchToProps } from "./DPAFCA.Props";

const mapStateToProps = (state: AppState): StateToProps => {
  const customer = Object.assign(
    new CustomerDetailModel(),
    state[namespace].customer.customer
  );
  return {
    dateOfBirth: Shared.getFormattedDate(customer.dateOfBirth, "DD/MM/YYYY"),
    address: customer.address,
  };
};

const mapDispatchToProps = (dispatch: Dispatch<any>): DispatchToProps => ({});

export default connect(mapStateToProps, mapDispatchToProps)(DPAFCA);
