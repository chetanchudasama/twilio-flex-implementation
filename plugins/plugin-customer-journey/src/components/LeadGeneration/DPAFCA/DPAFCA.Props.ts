export interface StateToProps {
  address: string;
  dateOfBirth: string;
}

export interface DispatchToProps {}

export type DPAFCAProps = StateToProps & DispatchToProps;
