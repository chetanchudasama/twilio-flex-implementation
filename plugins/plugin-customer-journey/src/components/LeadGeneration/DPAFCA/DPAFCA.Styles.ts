import styled from "react-emotion";
import {
  white,
  black,
  darkGrey,
  lightGrey,
  secondary,
} from "@common/components";

export const DPAFCAStyles = styled("div")({
  background: white,
  padding: 24,
  marginLeft: -8,
  color: darkGrey,
  ".heading": {
    fontSize: 20,
    lineHeight: "24px",
    fontWeight: 600,
    color: black,
  },
  ".dpafca-title": {
    fontSize: 16,
    fontWeight: 600,
    color: black,
  },
  ".date-of-birth": {
    display: "flex",
    marginBottom: 8,
  },
  ".dob-value": {
    marginTop: 1,
  },
  ".address": {
    borderBottom: `1px solid ${lightGrey}`,
    paddingTop: "0 !important",
    display: "flex",
  },
  ".icon": {
    marginRight: 12,
    color: secondary,
    opacity: 0.4,
  },
  ".icon svg": {
    height: 20,
    width: 16,
  },
  ".fca-note": {
    paddingTop: "0 !important",
  },
});
