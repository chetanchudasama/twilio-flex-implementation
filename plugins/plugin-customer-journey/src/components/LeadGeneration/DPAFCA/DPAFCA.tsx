import React from "react";
import { Grid } from "@material-ui/core";
import CakeIcon from "@material-ui/icons/Cake";
import HomeIcon from "@material-ui/icons/Home";

import { DPAFCAStyles } from "./DPAFCA.Styles";
import { DPAFCAProps } from "./DPAFCA.Props";

export const DPAFCA: React.FC<DPAFCAProps> = ({ address, dateOfBirth }) => {
  return (
    <DPAFCAStyles>
      <Grid container spacing={16}>
        <Grid item xs={12} className="heading">
          DPA & FCA
        </Grid>
        <Grid item xs={12} className="dpafca-title">
          DPA
        </Grid>
        <Grid item xs={12} className="address">
          <span className="icon">
            <HomeIcon />
          </span>
          <span>{address}</span>
        </Grid>
        <Grid item xs={12} className="date-of-birth">
          <span className="icon">
            <CakeIcon />
          </span>
          <span className="dob-value">{dateOfBirth}</span>
        </Grid>
        <Grid item xs={12} className="dpafca-title">
          FCA Statement
        </Grid>
        <Grid item xs={12} className="fca-note">
          Just to make you aware that all our calls are recorded for training
          and quality purposes and we are authorised and regulated by the
          Financial Conduct Authority.
        </Grid>
      </Grid>
    </DPAFCAStyles>
  );
};
