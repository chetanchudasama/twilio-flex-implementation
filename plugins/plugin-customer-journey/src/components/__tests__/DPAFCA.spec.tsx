import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";

import { DPAFCAProps } from "../LeadGeneration/DPAFCA/DPAFCA.Props";
import { DPAFCA } from "../LeadGeneration/DPAFCA/DPAFCA";

const props: DPAFCAProps = {
  address: "114, Parsley, Parsley-9, 48 Bullwood Rd, ST ARVANS, NP6 8RF",
  dateOfBirth: "16/05/1975",
};

describe("DPAFCA", () => {
  it("renders correctly on load", () => {
    const wrapper = create(<DPAFCA {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("should render address correctly", () => {
    const wrapper = mount(<DPAFCA {...props} />);

    expect(wrapper.find(".address").last().html()).toContain(props.address);
  });

  it("should render date of birth correctly", () => {
    const wrapper = mount(<DPAFCA {...props} />);

    expect(wrapper.find(".date-of-birth").last().html()).toContain(
      props.dateOfBirth
    );
  });
});
