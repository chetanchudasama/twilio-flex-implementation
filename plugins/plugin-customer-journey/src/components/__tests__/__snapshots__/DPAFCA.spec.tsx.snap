// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DPAFCA renders correctly on load 1`] = `
<div
  className="css-19v75dd"
>
  <div
    className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 heading"
    >
      DPA & FCA
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 dpafca-title"
    >
      DPA
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 address"
    >
      <span
        className="icon"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-98"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span>
        114, Parsley, Parsley-9, 48 Bullwood Rd, ST ARVANS, NP6 8RF
      </span>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 date-of-birth"
    >
      <span
        className="icon"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-98"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 6c1.11 0 2-.9 2-2 0-.38-.1-.73-.29-1.03L12 0l-1.71 2.97c-.19.3-.29.65-.29 1.03 0 1.1.9 2 2 2zm4.6 9.99l-1.07-1.07-1.08 1.07c-1.3 1.3-3.58 1.31-4.89 0l-1.07-1.07-1.09 1.07C6.75 16.64 5.88 17 4.96 17c-.73 0-1.4-.23-1.96-.61V21c0 .55.45 1 1 1h16c.55 0 1-.45 1-1v-4.61c-.56.38-1.23.61-1.96.61-.92 0-1.79-.36-2.44-1.01zM18 9h-5V7h-2v2H6c-1.66 0-3 1.34-3 3v1.54c0 1.08.88 1.96 1.96 1.96.52 0 1.02-.2 1.38-.57l2.14-2.13 2.13 2.13c.74.74 2.03.74 2.77 0l2.14-2.13 2.13 2.13c.37.37.86.57 1.38.57 1.08 0 1.96-.88 1.96-1.96V12C21 10.34 19.66 9 18 9z"
          />
        </svg>
      </span>
      <span
        className="dob-value"
      >
        16/05/1975
      </span>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 dpafca-title"
    >
      FCA Statement
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 fca-note"
    >
      Just to make you aware that all our calls are recorded for training and quality purposes and we are authorised and regulated by the Financial Conduct Authority.
    </div>
  </div>
</div>
`;
